src/ImportPage.js

```diff
@@ -19,27 +22,92 @@ type Props = {|
 |};
 
 export default function ImportPage({onDataImported}: Props) {
+  const processTimeline = useCallback(
+    (events: TimelineEvent[]) => {
+      // Filter null entries and sort by timestamp.
+      // I would not expect to have to do either of this,
+      // but some of the data being passed in requires it.
+      events = events.filter(Boolean).sort((a, b) => (a.ts > b.ts ? 1 : -1));
+
+      if (events.length > 0) {
+        const processedData = preprocessData(events);
+        const processedFlamechart = preprocessFlamechart(events);
+        onDataImported(processedData, processedFlamechart);
+      }
+    },
+    [onDataImported],
+  );
+
+  // ImportPage is show in production build only
   useEffect(() => {
+    if (process.env.NODE_ENV === 'production') return;
     fetch(JSON_PATH)
       .then(res => res.json())
-      .then((events: TimelineEvent[]) => {
-        // Filter null entries and sort by timestamp.
-        // I would not expect to have to do either of this,
-        // but some of the data being passed in requires it.
-        events = events.filter(Boolean).sort((a, b) => (a.ts > b.ts ? 1 : -1));
-
-        if (events.length > 0) {
-          const processedData = preprocessData(events);
-          const processedFlamechart = preprocessFlamechart(events);
-          onDataImported(processedData, processedFlamechart);
-        }
-      });
-  }, []);
+      .then(processTimeline);
+  }, [onDataImported, processTimeline]);
+
+  const inputProfilerData = useCallback(async event => {
+    const regex = /^.*\.json$/g;
+    const inputFile = event.target.files[0];
+    if (!inputFile.name.match(regex)) {
+      console.error(
+        'Invalid file type, insert a captured performance profile JSON',
+      );
+      return;
+    }
+
+    const readFile = await readInputData(inputFile);
+    processTimeline(JSON.parse(readFile));
+  });
 
   return (
-    <div>
-      LOADING. TODO: Turn this into an import page. This page currently just
-      immediately loads a JSON file.
+    <div className={style.App}>
+      <div className={style.container}>
+        <div className={style.card}>
+          <div className={style.cardcontainer}>
+            <div className={style.row}>
+              <div className={style.column}>
+                <img src={logo} className={style.reactlogo} alt="logo" />
+              </div>
+              <div className={style.columncontent}>
+                <h2>React Concurrent Mode Profiler</h2>
+                <hr />
+                <p>
+                  To use the scheduler-profiler, load a pre-captured{' '}
+                  <a
+                    className={style.Link}
+                    href="https://developers.google.com/web/tools/chrome-devtools/evaluate-performance">
+                    performance profile
+                  </a>{' '}
+                  from browser devtools.
```

Comment:
I think this can be clearer. Just a bunch of nits:

1. The first clause is a little redundant as the page title already mentions the product name and this sentence already sounds like an instruction
1. s/load/import to match our button text
1. ~pre-~ captured, as "captured" is already past tense
1. Let's specify the browser, since we're only supporting Chrome for now?
1. Capitalized "devtools"

```suggestion
                  Import a profile{' '}
                  <a
                    className={style.Link}
                    href="https://developers.google.com/web/tools/chrome-devtools/evaluate-performance">
                    captured
                  </a>{' '}
                  using the Chrome DevTools Performance panel.
```